test/unit/specs/core/createEventManager.spec.js

```diff
@@ -108,31 +108,51 @@ describe("createEventManager", () => {
         });
     });
 
-    it("sets the lastChanceCallback, which wraps config.onBeforeEventSend, on the event", () => {
-      let wrappedLastChanceCallback;
-      event.setLastChanceCallback.and.callFake(callback => {
-        wrappedLastChanceCallback = callback;
+    it("events call onBeforeEventSend callback", () => {
```

Comment:
I think this test is mainly just testing your test, since your event mock is the one actually calling `onBeforeEventSend` and then you're asserting that `onBeforeEventSend` was called. Instead, from a unit perspective, I think here you just want test that `event.sendEvent` calls `event.finalize` with `onBeforeEventSend`. The test for whether `onBeforeEventSend` actually gets called by `event.finalize` belongs in `createEvent.spec.js` instead. Let me know if this doesn't make sense.